client/layout/guided-tours/config-elements.js

```diff
@@ -103,10 +105,25 @@ export class Step extends Component {
 
 	componentWillMount() {
 		this.skipIfInvalidContext( this.props, this.context );
+		this.scrollContainer = query( this.props.scrollContainer )[ 0 ] || global.window;
+		this.setStepPosition( this.props );
+	}
+
+	componentDidMount() {
+		global.window.addEventListener( 'resize', this.onScrollOrResize );
 	}
 
 	componentWillReceiveProps( nextProps, nextContext ) {
 		this.skipIfInvalidContext( nextProps, nextContext );
+		this.scrollContainer = query( nextProps.scrollContainer )[ 0 ] || global.window;
+		this.scrollContainer.addEventListener( 'scroll', this.onScrollOrResize );
```

Comment:
When props change, we reassign `this.scrollContainer` and bind `onScrollOrResize` to that potentially new DOM node; however, `removeEventListener` is only invoked when unmounting. Seems like the sequence here should be:

```js
this.scrollContainer.removeEventListener( 'scroll', this.onScrollOrResize );
this.scrollContainer = query( nextProps.scrollContainer )[ 0 ] || global.window;
this.scrollContainer.addEventListener( 'scroll', this.onScrollOrResize );
```